### ISSUE TITLE
TripItPro Status Field

### ISSUE THREAD

**Issue by azampagl** [tripit/api#238]
In regards to the status field in I have followed the documentation have successfully created trip and included the Status field.

For some reason, all of the Segments are saying "Missing Info" where the status usually is on both Android and iOS TripIt Pro accounts (this does not occur on normal accounts).

This is an example API post, which includes the Status (AirUpdate).

`<?xml><AirObject><trip_id>282740115</trip_id><booking_site_name>MySite.com</booking_site_name><supplier_email_address>info@chartermatrix.com</supplier_email_address><supplier_name>DEMO</supplier_name><supplier_phone>1.888.777.7777</supplier_phone><notes>\r\nCrew\r\n-------------------\r\n\r\nMatt O'Brien II (PIC) - 1111111111\r\n\r\n</notes><Segment><Status><flight_status>200</flight_status><last_modified>1568043770</last_modified></Status><StartDateTime><date>2019-09-09</date><time>13:14:00</time></StartDateTime><EndDateTime><date>2019-09-09</date><time>20:43:16</time></EndDateTime><start_airport_code>LAX</start_airport_code><start_city_name>Los Angeles</start_city_name><end_airport_code>TEB</end_airport_code><end_city_name>TETERBORO</end_city_name><marketing_airline>DEMO</marketing_airline><marketing_airline_code>AM</marketing_airline_code><marketing_flight_number>2020Z-30</marketing_flight_number><operating_airline>DEMO</operating_airline><operating_airline_code>AM</operating_airline_code><operating_flight_number>2020Z-30</operating_flight_number><aircraft>N555RG</aircraft><aircraft_display_name>N555RG</aircraft_display_name><distance>2454 miles</distance><stops>nonstop</stops></Segment></AirObject>`

The responses is successful with no errors, yet the app says "Missing Info".

Please see the attached screenshot for what the user sees.
https://imgur.com/a/Y4xNHV2

**Comment by hliss** [comment]
Hiya,

So, "Missing Info" means that TripIt needs more (or corrected) information to try to monitor the flight. The baseline data that's required is: 

- marketing airline code
- marketing flight number
- departure date
- origin airport code
- destination airport code

It looks like the issue here is likely with the airline/flight-number combination. Marketing airline code is a read-only field, so the marketing airline of `DEMO` wouldn't work, and the flight number of `2020Z-30` is invalid. You should send the IATA code in `marketing_airline`, and a `flight_number` should be a one to four digit number.

Please also keep in mind that private/general aviation (as would fly into, say, Teterboro) would generally not be monitorable. Hopefully this helps you narrow down the error.

**Comment by hliss** [comment]
I'll add one more bit because I didn't address the fact you mentioned that the segment was successfully saved in TripIt. TripIt is, as you've noted, pretty permissive in what is accepted in all of these fields. There's no reason to reject this data being your itinerary, but more information would be required to monitor the status of your flight.

**Comment by azampagl** [comment]
@hliss Thanks for the info. I've tried every combination of providing marketing_airline, marketing_airline_code, marketing_flight_number (in addition to operatoring_*) with your suggestions (IATA code in the marketing airline, only numeric flight numbers) - no success. The pro version continues to show missing info. I'm trying to set the status of "Not Monitored" (status = 200).

See: https://imgur.com/a/u5fEORg

Any thoughts?

AirObject IDs: 1079259089   1079263611 if you need it for reference.

**Comment by hliss** [comment]
Hiya,

`Missing Info` will show up when one of those five fields is empty. Let's see if we can get you past the error. You should set `marketing_airline` to an IATA code or the airline name, preferably the IATA code. `marketing_airline_code` is read-only and will be ignored. `marketing_flight_number` is also required for this purpose. 

Once those fields are filled out, and critically the airline is resolved to a real known airline, `Missing Info` will go away. I'm not sure TripIt would be able to monitor an LAX-TEB charter flight, but you'd at least get past this behavior.

**Comment by azampagl** [comment]
@hliss Yup, I set marketing_airline to an IATA code (LAX) and the marketing_flight_number to "1" and it still shows missing info.

If I set the marketing_airline to an IATA code, how is it suppose to "resolve to a real known airline"?

**Comment by hliss** [comment]
@azampagl Try an airline IATA code such as AA, UA, DL, BA, LH in `marketing_airline`?

**Comment by azampagl** [comment]
@hliss I'm not sure that is something we want to do. I'm adding flights that are private/corporate/charter related and we know the status - I was hoping to be able to control the status field and not have be based on the actual flights of commercial airlines.

**Comment by azampagl** [comment]
@hliss Is there any marketing_airline that we can put that lets the developer control the status? E.g. marketing_airline = XX

**Comment by hliss** [comment]
@azampagl Aha! Gotcha, now I'm caught up with what you'd like to do. Sadly, no, all status fields are currently treated as read-only via the API. The XSD specifies that `All FlightStatus fields are read-only and only available for monitored TripIt Pro AirSegments`. 

However, I think this is a fantastic feature request for us to consider! This hadn't been our primary use case, of course, but I kind of like the idea of a developer being able to push and own their own flight status if the flight is not commercially monitorable. I wouldn't expect it to be available anytime in the near future, so I would guide you not to plan for anything, but let's get it under consideration. Thanks!!

**Comment by azampagl** [comment]
@hliss Just wanted to check up on the status of controlling the flight status. My company has some potentially big integration questions we'd like to discuss with you if you could email me at azampagl@daitollc.com

**Comment by hliss** [comment]
Hiya, hope you're well. We are still indeed tracking the feature request but I don't have any other updates. Feel free to ask the integration questions you mentioned to api-support@tripit.com!
